packages/commerce-sdk-react/src/provider.tsx

```diff
@@ -17,8 +20,33 @@ export interface CommerceAPIProviderProps {
     currency: string
 }
 
+export const CommerceAPIContext = React.createContext({} as ApiClients)
+
+// TODO: how to test? test in typescript template for now
 const CommerceAPIProvider = (props: CommerceAPIProviderProps): ReactElement => {
-    return <></>
+    const {children, clientId, organizationId, shortCode, siteId, proxy} = props
+
+    const config = {
+        proxy,
+        headers: {},
+        parameters: {
+            clientId,
+            organizationId,
+            shortCode,
+            siteId,
+        },
+        throwOnBadResponse: true,
+    }
+
+    // TODO: auth logic should use the helpers from commerce-sdk-isomorphic ?
+    const commerceAPI = new CommerceAPI(config)
+
+    console.log('--- api client', commerceAPI)
+
+    // TODO: use Context from useServerEffect
+    // See Kevin's PR: https://github.com/SalesforceCommerceCloud/pwa-kit/pull/654/files#r914097886
+    // See Ben's PR: https://github.com/SalesforceCommerceCloud/pwa-kit/pull/642
+    return <CommerceAPIContext.Provider value={commerceAPI}>{children}</CommerceAPIContext.Provider>
```

Comment:
Ticket's acceptance criteria ask for integrating the `useServerEffect`. But since that work is still in progress and done in parallel, I can't integrate it here right now.